apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.component.ts

```diff
@@ -0,0 +1,138 @@
+import { ChangeDetectionStrategy, Component, computed, effect, type OnDestroy, type Signal, ViewEncapsulation } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { DfTooltipModule } from '@design-factory/design-factory';
+import { NgbTypeaheadModule } from '@ng-bootstrap/ng-bootstrap';
+import { GetStylingVariableContentMessage, StylingVariables } from '@o3r/styling';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, map, shareReplay, startWith, throttleTime } from 'rxjs/operators';
+import { ChromeExtensionConnectionService } from '../../services/connection.service';
+import { ColorPipe } from './color.pipe';
+import { resolveVariable, searchFn, type Variable } from './common';
+import { IsRefPipe } from './is-ref.pipe';
+import { MemoizePipe } from './memoize.pipe';
+import { VariableLabelPipe } from './variable-label.pipe';
+
+// TODO define it in @o3r/styling or @o3r/design
+const THEME_CATEGORY_NAME = 'theme';
+
+const THROTTLE_TIME = 100;
+
+@Component({
+  selector: 'o3r-theming-panel-pres',
+  templateUrl: './theming-panel-pres.template.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  standalone: true,
+  imports: [
+    IsRefPipe,
+    ReactiveFormsModule,
+    FormsModule,
+    ColorPipe,
+    NgbTypeaheadModule,
+    VariableLabelPipe,
+    DfTooltipModule,
+    MemoizePipe
+  ]
+})
+export class ThemingPanelPresComponent implements OnDestroy {
+  public readonly resolvedVariables: Signal<Record<string, string>>;
+  public readonly numberOfVariables: Signal<number>;
+  public readonly themeVariables: Signal<StylingVariables>;
+  public readonly filteredThemeVariables: Signal<StylingVariables>;
+  public readonly form = new FormGroup({
+    variables: new FormGroup<Record<string, FormControl<string | null>>>({}),
+    search: new FormControl('')
+  });
+
+  private readonly variables$: Observable<StylingVariables>;
+  private readonly subscription = new Subscription();
+  private readonly runtimeValues$ = this.form.controls.variables.valueChanges.pipe(startWith({}));
+  private readonly runtimeValues = toSignal(this.runtimeValues$, { initialValue: {} });
+
+  constructor(
+    connectionService: ChromeExtensionConnectionService
+  ) {
+    connectionService.sendMessage(
```

Comment:
As the JS run is faster than a postmessage it should not be an issue but you are requesting the information before be plugged to the service to receive it :)